### DISCUSSION TITLE
Day & Night for same relay

### DISCUSSION THREAD

**Discussion by threedom** [theyosh/TerrariumPI#795] [Q&A] [answered]
Hi everyone,

is there a way for me to run a relay in a different mode during the day than during night? 
My specific use case is I have a waterfall that I want to run permanently during the day (7am-7pm). However, since there is moss on the waterfall that I don't want it to dry out at night, I'd like to run it for 10 minutes every hour during the night (8pm -6am). 

So generally the settings make sense to me but it seems like I can't have 
a) have day and night settings in the same area
b) have the same relay in two areas.

Is there any way that I am missing in which I could do what I need to do?

Thanks in advance!

**Comment by theyosh** [comment]
Hi,

short answer: no, not with the same relay.

long answer: There are ways of doing this.
I think this could be a solution: https://www.buildmyowncabin.com/electrical/how-to-wire-3-way-switch.html
You just add a new relay what is also controlling the waterfall. But I would ask an electronic handy friend to see if this can be done safely.

The only downside is, that you have to make sure there is no overlapping in the period the relays are on. As 1 relay will put it on, the other relay can put it off, when it thinks it turns it on.

In you example there are 1 hour between the 'change' 7am-7pm and then 8pm -6am. So that should be safe.

**Reply by theyosh** [reply, marked as answer]
Hmm, looks like this will not work with the relays I can control... It needs 3 wire relays.. not sure if that will work.. :(

**Reply by threedom** [reply]
Thanks for the reply. 
I think tinkering with the wiring too much would be a bit too risky for my purposes. 
Would this qualify as a new idea / feature request for future use? Maybe it only seems like a useful feature to me. My use case may be specific where I want the waterfall to stop most of the night to 
A) minimize the risk of leaks
B) minimally disturb my snake
But also need it to run for brief intervals to ensure that the live moss on it doesn't dry out too much. 

As a sidenote: The way I tried to solve it is by using the kasa app and just giving it instructions for the night time routine separately from terrariumpi, however, terrariumpi shuts that down once it notices that the plug is on and shouldn't be. So that doesn't work perfectly, but at least the moss gets 30-60s of water before terrariumpi shuts it down.
